specs/MultiProfile.md

```diff
@@ -363,16 +384,36 @@ interface ICoreWebView2Profile2 : ICoreWebView2Profile {
   [propget] HRESULT CookieManager([out, retval] ICoreWebView2CookieManager** cookieManager);
 }
 
-[uuid(1c1ae2cc-d5c2-ffe3-d3e7-7857035d23b7), object, pointer_default(unique)]
-interface ICoreWebView2Profile3 : ICoreWebView2Profile2 {
-  /// All webviews on this profile will be closed, and the profile will be marked for deletion.
-  /// After the Delete() call completes, The render process of webviews on this profile will
-  /// asynchronously exit with the reason:`COREWEBVIEW2_PROCESS_FAILED_REASON_PROFILE_DELETED`.
-  /// See 'COREWEBVIEW2_PROCESS_FAILED_REASON::COREWEBVIEW2_PROCESS_FAILED_REASON_PROFILE_DELETED'
-  /// for more details. The profile directory on disk will be actually deleted when the browser
-  /// process exits. Webview2 creation will fail with the HRESULT is ERROR_INVALID_STATE(0x8007139FL)
-  /// if you create it with the same name as a profile that is being deleted.
-  HRESULT Delete(); 
+[uuid(2765B8BD-7C57-4B76-B8AA-1EC940FE92CC), object, pointer_default(unique)]
+interface ICoreWebView2StagingProfile4 : IUnknown {
+  /// After the API is called, the profile will be marked for deletion. The
+  /// local profile’s directory will be tried to delete at browser process
+  /// exit, if fail to delete, it will recursively try to delete at next
+  /// browser process start until successful.
+  /// The corresponding user's `ProfileDeleted` event handle function will
+  /// be triggered. After the function is triggered, continuing to use the
+  /// profile or its corresponding webviews is an undefined behavior.
+  /// For each WebView of the same profile, the WebView will be auto closed.
+  /// If the user has registered an event handler for ProfileDeleted event,
+  /// the event handle will be invoked before WebView is closed.
+  /// If create a new profile with the same name as the profile that has been
+  /// marked as deleted will be failure with the HRESULT:ERROR_INVALID_STATE
+  /// (0x8007139FL).
+  HRESULT Delete();
+}
+
+[uuid(cc39bea3-f6f8-471b-919f-fa253e2fff03), object, pointer_default(unique)]
+interface ICoreWebView2Staging9 : IUnknown {
+  /// The `ProfileDeleted` event is raised when its corresponding Profile's
+  /// Delete API is called. When this event has been raised, continue to use
```

Comment:
Also, if this event is raised before the profile is deleted and before the profile and webview2 objects are closed, we should have an event args class for this event with a GetDeferral method on it (following the example of other GetDeferral methods we have in WebView2). The deferral would defer the actual deletion of the profile and closing of the webview2 until the deferral is completed. This would allow the caller to perform asynchronous work as you might expect be required if they need to interact with UI.

Reply:
Yes, this is Option 2 in the design doc(https://microsoft.sharepoint.com/:w:/r/teams/Edge/_layouts/15/Doc.aspx?sourcedoc=%7B63768506-3801-4625-A15C-69E43F68C2DC%7D&file=Profile%27s%20Delete%20API%20redesign.docx&action=default&mobileredirect=true&share=IQEGhXZjATglRqFcaeQ_aMLcAQl8rWiHxurWk807-8BWjYU&cid=025aac02-f7da-4e6a-88bd-c2eb8100bddc), but we think let the user continue to use the webview will delay the webview's close. If we really want to let the use can continue to use the webview in the event handle, it is better to let the user to close the webview, this is the Option 3 in the design doc. Taken together, we have chosen the option 1: just not allow the user to continue to use the webview2 upon the event handle is raised.

Reply:
Based on what you said before this event doesn't block anything so we don't need a GetDeferral.